keyboards/planck/keymaps/sigul/keymap.c

```diff
@@ -0,0 +1,384 @@
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |  F6  |  F7  |  F8  |  F9  | F10  |  F1  |  F2  |  F3  |  F4  | F5   |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      | RAISE|      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+
+[_FN] = LAYOUT_ortho_4x12(
+    _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,      KC_F7,      KC_F8,   KC_F9,    KC_F10,  KC_DEL,
+    _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT,    KC_DOWN,    KC_UP,   KC_RIGHT, _______, _______,
+    _______, KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F1,      KC_F2,      KC_F3,   KC_F4,    KC_F5,   _______,
+    _______, _______, _______, _______, _______, _______, _______,    TG(_RAISE), _______, _______, _______,  _______
+),
+
+/* MOUSE
+ * ,-----------------------------------------------------------------------------------.
+ * |      |      |      |      |      |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |      |      |      |      | MOUSE|      |      |      |Scr Up|Scr Do|      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |But1  |      |      |But2  | Left | Down | Up   |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+
+[_MOUSE] = LAYOUT_ortho_4x12(
+    _______, _______, _______, _______, _______,    _______, _______, _______,    _______,    _______,       _______,     _______,
+    _______, _______, _______, _______, _______,    _______, _______, _______,    _______,    _______,       _______,     _______,
+    _______, _______, _______, _______, _______,    MOUSE,   _______, _______,    _______,    KC_MS_WH_DOWN, KC_MS_WH_UP, _______,
+    _______, _______, _______, _______, KC_MS_BTN1, _______, _______, KC_MS_BTN2, KC_MS_LEFT, KC_MS_DOWN,    KC_MS_UP,    KC_MS_RIGHT
+)
+
+};
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+  
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
```

Comment:
If you're only using QWERTY, then you don't need this, and you may want to remove it outright. 

```suggestion
```

Reply:
I understand, but I find it useful when for any reason I mess up with the layers to get back to the default.

Reply:
Well, this won't affect the other layers. It only affects the default layer.  So, it doesn't do anything.

Though, `TO(0)` may be helpful, as this turns off all of the layers, and moves to the default layer.

Reply:
so this should became 

    #define QWERTY TO(_QWERTY)

in definitions (I moved them in users/sigul/sigul.h as I am working to make the secrets)